cognite/neat/graph/transformers/_value_type.py

```
@@ -0,0 +1,110 @@
+                        traversal.property.suffix
+                    ]
+
+                    subject_uri = rules.prefixes.get(traversal.class_.prefix, rules.metadata.namespace)[
+                        traversal.class_.suffix
+                    ]
+
+                    # needs forming object_uri this way since we are also
+                    # translating original rdf types to rdf types in new
+                    # namespace so we need to pick the correct one which are
+                    # store in class references
+                    if isinstance(value_type, ClassEntity) and (
+                        object_uri := class_reference_pairs.get(value_type, None)
+                    ):
+                        for s, o in graph.query(  # type: ignore [misc]
+                            self._object_property_template.format(
+                                subject_uri=subject_uri,
+                                property_uri=property_uri,
+                                object_uri=object_uri,
+                            )
+                        ):
+                            graph.remove((s, property_uri, o))
+                            new_property = URIRef(f"{property_uri}_{value_type.suffix}")
+                            graph.add((s, new_property, o))
+
+                    elif isinstance(value_type, DataType) and not isinstance(value_type, AnyURI):
+                        for s, o in graph.query(  # type: ignore [misc]
+                            self._datatype_property_template.format(
+                                subject_uri=subject_uri,
+                                property_uri=property_uri,
+                                datatype=XSD[value_type.xsd],
+                            )
+                        ):
+                            graph.remove((s, property_uri, o))
+                            new_property = URIRef(f"{property_uri}_{value_type.xsd}")
+                            graph.add((s, new_property, o))
```

Not sure if it is worth it, but warning of error there suddenly is something unexpected. 

The motivation for raising an error for a case that should not ever happen, is that in case it actually does you get a nice error message instead of something which is typically very hard to understand or you avoid silently failing. 

```suggestion
                            graph.add((s, new_property, o))
                    else:
                        raise ValueError(f"Value type {value_type} is not supported for splitting multi-value properties")
```